bin/cache.cpp

```diff
@@ -0,0 +1,104 @@
+#include <mbgl/storage/default_file_source.hpp>
+#include <mbgl/storage/resource.hpp>
+#include <mbgl/util/chrono.hpp>
+#include <mbgl/util/run_loop.hpp>
+
+#include <args.hxx>
+
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+std::shared_ptr<std::string> readFile(const std::string& fileName) {
+    std::ifstream stream(fileName.c_str());
+
+    if (!stream.good()) {
+        std::cerr << "Error: cannot read file " + fileName << std::endl;
+        exit(1);
+    }
+
+    std::stringstream buffer;
+    buffer << stream.rdbuf();
+
+    return std::make_shared<std::string>(buffer.str());
+}
+
+int main(int argc, char* argv[]) {
+    args::ArgumentParser p("Mapbox GL cache tool", "");
+    args::HelpFlag helpFlag(p, "help", "Display this help menu", {'h'});
+
+    args::ValueFlag<std::string> urlValue(p, "URL", "Resource URL (required)", {'u'}, args::Options::Required);
+    args::ValueFlag<std::string> cacheValue(
+        p, "cache", "Path to the cache database (required)", {'c'}, args::Options::Required);
+    args::ValueFlag<std::string> dataValue(
+        p, "data", "Path to the resource data (required)", {'d'}, args::Options::Required);
+    args::ValueFlag<std::string> etagValue(p, "etag", "Cache eTag, none otherwise", {'t'});
+    args::ValueFlag<unsigned long> expiresValue(p, "expires", "Expires date, will use 'now' otherwise", {'e'});
+    args::ValueFlag<unsigned long> modifiedValue(p, "modified", "Modified date, will use 'now' otherwise", {'m'});
+
+    std::unordered_map<std::string, mbgl::Resource::Kind> typeMap{{"glyphs", mbgl::Resource::Glyphs},
```

Comment:
`std::map` is enough

Reply:
```
    template <
        typename K,
        typename T,
        typename Reader = ValueReader,
        template <typename...> class Map = std::unordered_map>
    class MapFlag : public ValueFlagBase
    {    
```

It is used in the `MapFlag` that only takes ordered. I get compilation errors when changing to `std::map` :-(